syntax = "proto3";

package queue;
option go_package = "github.com/JuanGQCadavid/ds-practice-2025/utils/pb/order_queue";

import "common/common.proto";

service OrderQueueService {
    rpc enqueue (EnqueueRequest) returns (common.NextResponse);
    rpc dequeue (EmptyRequest) returns (DequeueResponse);
    rpc clean (EmptyRequest) returns (common.NextResponse);
}

message EmptyRequest {}

message EnqueueRequest {
    string orderId = 1;
    common.Order order = 2;
}

message EnqueueResponse {
    string errMessage = 1;
    bool isValid = 2;
}

message DequeueResponse {
    string orderId = 1;
    common.Order order = 2;
    string errMessage = 3;
    bool isValid = 4;
}